src/main/java/com/zs/hobbytracker/Hobby.java

```diff
@@ -0,0 +1,172 @@
+package com.zs.hobbytracker;
+
+import com.zs.hobbytracker.controller.BadmintonController;
+import com.zs.hobbytracker.controller.ChessController;
+import com.zs.hobbytracker.controller.UserController;
+import com.zs.hobbytracker.exception.InvalidInputException;
+import com.zs.hobbytracker.lru.Cache;
+import com.zs.hobbytracker.lru.LruCache;
+import com.zs.hobbytracker.models.Badminton;
+import com.zs.hobbytracker.models.Chess;
+import com.zs.hobbytracker.utils.DatabaseConnection;
+
+import java.sql.Connection;
+import java.sql.Date;
+import java.util.InputMismatchException;
+import java.util.Scanner;
+import java.util.logging.Logger;
+
+
+/**
+ * Following class is the entry point for the program of hobbies. User may have multiple hobbies.
+ * Users can know their longest streak hobby wise.
+ * Users can know their recent streak hobby wise.
+ * Users can know when they recently performed hobby.
+ * Users can fetch the details hobby wise on any given date
+ */
+public class Hobby {
+
+
+    public static final Logger logger = com.zs.hobbytracker.logger.Logger.getLogger();
+    public static LruCache lruCache;
+    public static Cache cache;
+
+    /**
+     * Functionality for hobby
+     */
+    public static void hobby() {
+        Connection connection = DatabaseConnection.getConnection();
+        if (connection == null) {
+            logger.severe("Connection not established");
+            logger.info("Not Connected");
+        } else {
+            Scanner scanner = new Scanner(System.in);
```

Comment:
Make controllers class level variables and also initialize in a different method.